Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift

```diff
@@ -241,10 +295,27 @@ internal protocol SpecTest: Decodable {
 
     /// List of expected CommandStartedEvents.
     var expectations: [TestCommandStartedEvent]? { get }
+
+    /// Document describing the return value and/or expected state of the collection after the operation is executed.
+    var outcome: TestOutcome? { get }
+
+    /// Map of session names (e.g. "session0") to parameters to pass to `MongoClient.startSession()` when creating that
+    /// session.
+    var sessionOptions: [String: ClientSessionOptions]? { get }
+
+    /// Array of session names (e.g. "session0", "session1") that the test refers to. Each session is proactively
+    /// started in `run()`.
+    static var sessionNames: [String] { get }
```

Comment:
There can only be sessions that exist here. The reason for the dictionary of session names (i.e. "session0") to `ClientSession`s is that the spec test assert that `CommandStartedEvent`s contain the field `"lsid": session0`. Since the `lsid` is generated automatically by the server and cannot be specified, we need to check that the generated `lsid` corresponds to that of "session0".

Reply:
We've added this field so that you can specify only the session names that a spec test refers to. For instance, one test may refer to "session0" and "session1" whereas another might refer to "sessionA" and "sessionB". This gives us the flexibility to reflect any of these changes down the line.